newIDE/app/src/VariablesList/useRefocusField.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+import React from 'react';
+
+const useRefocusField = (fieldRefs: {|
+  current: {|
+    [identifier: number]: {|
+      +focus: (?{| caretPosition: ?('end' | number) |}) => void,
+    |},
+  |},
+|}) => {
+  const [fieldToFocus, setFieldToFocus] = React.useState<?{
+    identifier: number,
+    caretPosition?: ?number,
+  }>(null);
+
+  React.useEffect(
+    () => {
+      if (fieldToFocus) {
+        const fieldRef = fieldRefs.current[fieldToFocus.identifier];
+        if (fieldRef) {
+          fieldRef.focus({ caretPosition: fieldToFocus.caretPosition });
+          setFieldToFocus(null);
+        }
+      }
+    },
+    // The fieldRefs is a reference, hence it does not need to be added as a dependency.
```

Comment:
This comment makes sense for a useCallBack() but unsure about a useEffect()
If for some reason the fieldRef change, the effect wouldn't be called again.
So maybe the comment should be: "we don't expect the field to change, so no need to add it as a dependency"
Or more "even if it changes, we only want to focus the field once" or something like that?